src/order/locally_finite.lean

```diff
@@ -482,6 +482,239 @@ locally_finite_order.of_Icc' (α × β)
 
 end preorder
 
+/-!
+#### `with_top`, `with_bot`
+
+Adding a `⊤` to a locally finite `order_top` keeps it locally finite.
+Adding a `⊥` to a locally finite `order_bot` keeps it locally finite.
+-/
+
+section order_top
+open with_top
+variables (α) [order_top α] [locally_finite_order α]
+
+instance : locally_finite_order (with_top α) :=
+{ finset_Icc := λ a b, match a, b with
+    |        ⊤,        ⊤ := {⊤}
+    |        ⊤, (some b) := ∅
```

Comment:
A bit: I don't think these parens around `some` are needed, as this is the version of pattern matching with commas.

Reply:
Eh, you're right. The discrepancies between usual pattern mathcing and match blocks utterly confuses me.